### ISSUE TITLE
Ally Switch in (two trainer) double battles

### ISSUE THREAD

**Issue by Dreadway420** [rh-hideout/pokeemerald-expansion#5638]
### Description

Problem: In a double battle (the type that occurs when two independent trainers see the player simultaneously), if a pokemon faints in the spot that is not 'owned' by its corresponding trainer after Ally Switch, then the next mon that is sent out will be from the trainer's party that 'owns' that spot. This is the behavior when one trainer has only pokemon. If both trainers have remaining pokemon in their party, it turns into a buggy mess with duplicate mons being sent out. Scenario 1 is simpler to explain, and I believe both have to do with Ally Switch. 

To reproduce:
Two independent trainers see the player, initiating a double battle.
Trainer 1: Clefairy with Ally Switch
Trainer 2: Pokemon X and Pokemon Y

Turn 1: Clefairy uses ally switch, they switch spots, X does whatever.

Turn 2: Clefairy faints, Pokemon Y will be sent out. Trainer 2 now has both mons on the field.

Expected behavior: Pokemon Y should not be sent out until Pokemon X faints. I am unsure if this works this way on cartridge, and I am unsure if there is a later gen dual trainer double battle with ally switch that could replicate this situation. 

My theory on one independent trainer 'owning' either the right or left side is purely speculative, as I am a novice and cannot locate where this is handled in the code. If this is a known issue or I am missing something, please let me know. I am happy to provide any other information as needed! 

### Version

1.9.3 (Latest release)

### Discord contact info

uvula_

**Comment by Dreadway420** [comment]
https://github.com/user-attachments/assets/d3c38caf-cb77-4354-82f0-3dc3683145ec

Correction: In the above explanation, the example clefairy can be the surviving mon, and a duplicate version of the clefairy will be sent out by trainer 2. The original clefairy was in the spot on the field that it did not start in when pokemon Y fainted. I have seen this happen where mon Y will be duplicated to the point that there are multiple (Durant in my case) stacked on top of each other. I have not been able to replicate this specific behavior yet.

In short: if a trainer has a surviving mon, even if it is one the field, and that trainer's original 'spot' is open, it will send out a duplicate of the mon that is on the field (currently in the other trainer's 'spot').
